libp2p/protocols/connectivity/autonatv2/utils.nim

```diff
@@ -0,0 +1,110 @@
+{.push raises: [].}
+
+import results, strutils
+import chronos
+import
+  ../../protocol,
+  ../../../switch,
+  ../../../multiaddress,
+  ../../../multicodec,
+  ../../../peerid,
+  ../../../protobuf/minprotobuf,
+  ../autonat/service,
+  ./types
+
+proc ipAddrMatches*(observedIPAddr: MultiAddress, reqAddrs: seq[MultiAddress]): bool =
+  ## Checks if at least one of the requests' addresses
+  ## has the same IP address of observedIPAddr
+
+  let observedIPAddr = observedIPAddr[0].valueOr:
```

Comment:
There's no guarantee that the first element of the address is gonna be an ip (i.e. `/p2p/somePeerId` is a valid multiaddress). Look into how observedaddrmanager extracts IP for inspiration. https://github.com/vacp2p/nim-libp2p/blob/master/libp2p/observedaddrmanager.nim#L30-L67